engine/classes/Elgg/Upgrade/Result.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Elgg\Upgrade;
+
+/**
+ * Result of a single BatchUpgrade run
+ */
+class Result {
```

Comment:
I assume this gets serialized between runs? Document that and maybe mark this `final` for now.